index.d.ts

```diff
@@ -0,0 +1,166 @@
+import * as React from "react";
+
+interface ProviderProps {
+  apiKey: string;
+  children?: React.ReactNode;
+}
+
+export class StripeProvider extends React.Component<ProviderProps, {}> {}
+
+export function injectStripe<Props extends Object>(
+  WrappedComponent: React.ComponentClass<Props & StripeProps>,
+): React.ComponentClass<Props>;
+
+export interface StripeProps {
+  stripe: StripeShape;
+}
+
+type CreateTokenResult = {
+  token: Token;
+  error: void;
+} | {
+  // TODO: StripeError can be more specific.
+  error: any;
```

Comment:
This should really be a `StripeError` definition from: https://stripe.com/docs/api#errors